tools/tests/test_backend_datadog.py

```diff
@@ -0,0 +1,132 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import yaml
+
+import unittest
+
+from sigma.configuration import SigmaConfiguration
+from sigma.parser.rule import SigmaParser
+from sigma.config.mapping import FieldMapping
+from sigma.backends.datadog import DatadogLogsBackend
+
+
+class TestDatadogLogsBackend(unittest.TestCase):
+    """Test cases for the Datadog Logs backend."""
+
+    def setUp(self):
+        self.basic_rule = {
+            "detection": {"selection": {"attribute": "test"}, "condition": "selection"}
+        }
+
+    def test_attribute(self):
+        query = self.generate_query(self.basic_rule)
+        expected_query = "@attribute:test"
+        self.assertEqual(query, expected_query)
+
+    def test_facets_backend_option(self):
+        query = self.generate_query(
+            self.basic_rule, backend_options={"index": "test_index"}
+        )
+        expected_query = "index:test_index AND @attribute:test"
+        self.assertEqual(query, expected_query)
+
+    def test_facets_config(self):
+        self.basic_rule["detection"]["selection"]["test-facet"] = "myfacet"
+        query = self.generate_query(self.basic_rule, config={"facets": ["test-facet"]})
+        expected_query = "@attribute:test AND test-facet:myfacet"
+        self.assertEqual(query, expected_query)
+
+    def test_special_characters_escape(self):
+        self.basic_rule["detection"]["selection"][
+            "regex-attribute"
+        ] = "anything?inbetween"
+        query = self.generate_query(self.basic_rule)
+        expected_query = "@attribute:test AND @regex-attribute:anything\\?inbetween"
+        self.assertEqual(query, expected_query)
+
+    def test_space_escape(self):
+        self.basic_rule["detection"]["selection"]["space-attribute"] = "with space"
+        query = self.generate_query(self.basic_rule)
+        expected_query = "@attribute:test AND @space-attribute:with?space"
+        self.assertEqual(query, expected_query)
+
+    def test_space_escape(self):
+        query = self.generate_query(
+            self.basic_rule,
+            fieldmappings={"attribute": FieldMapping("attribute", "another_attribute")},
+        )
+        expected_query = "@another_attribute:test"
+        self.assertEqual(query, expected_query)
+
+    def generate_query(
+        self, rule, backend_options=dict(), config=dict(), fieldmappings=dict()
+    ):
+        cfg = SigmaConfiguration()
+        cfg.config = config
+        cfg.fieldmappings = fieldmappings
+        backend = DatadogLogsBackend(cfg, backend_options)
+        parser = SigmaParser(rule, cfg)
+
+        return backend.generate(parser)
+
+    def test_all_sigma_rules(self):
```

Comment:
We need an assertion to make this a test.

IIUC we want to make sure it runs, but we don't specifically very the output?

I recommend we either assert specifically on the output, or we assert on the number of tests that passes.

Reply:
This is indeed not really a test. We moved it to a dedicated script logic and added explanations on how to run it in the README in 1712e9d.